### ISSUE TITLE
Remove extension declarations from CallMap and PropertyMap

### ISSUE THREAD

**Issue by AndrolGenhald** [vimeo/psalm#7514]
Now that we support enabling and disabling extensions based on composer.json and xml config, we'll need to remove all extension related info from CallMap and PropertyMap and move them to stubs so that they're correctly missing when the extension is disabled.

I'll open issues for individual extensions and list them here once #7513 is finished, since that should make the task much easier.

**Comment by orklah** [comment]
Yeah, we'll only be able to do that once the tool is created and stubs are available. 

Note: we'll have to go back to pretty old versions of PHP, there are project out there that use Psalm for 5.X codebases.

For very common extensions, we'll be able to use 3v4l to get the stub on old versions. For more rare extensions, we'll need to ask people to generate them and PR them from their environment

**Comment by AndrolGenhald** [comment]
> there are project out there that use Psalm for 5.X codebases.

:astonished: I didn't realize that. Yeah, 3v4l should be helpful for most, but I might try to set up some sort of semi-automated docker thing for the rest.

**Comment by orklah** [comment]
Yeah, officially, the callmap goes back to PHP 7.0 so there might be some differences, but for most things, it works :).

We wouldn't cut a lot of complexity by actually removing support for those so it'll probably stay for a long time (I'd say, those projects are possibly those who need Psalm the most!)
